### PR TITLE
blob: add Bucket.Close method

### LINKED ISSUE #1485: runtimevar/blobvar: buckets that are opened by URLOpener should be closed when the Variable is closed

### LINKED ISSUE #1492: blob: Bucket does not have a Close method
All of our other APIs have `Close` methods with the exception of `Bucket`. I don't think that we should constrain implementations of `Bucket` to not be able to hold long-lived connections, so I think bucket driver types and the portable type should support `Close`, even if all current implementations no-op.

(This would be a breaking change, but one that is fairly easy to correct for driver types.)

/cc @vangent for triage and assigning.

### PR DESCRIPTION
**Pull request by zombiezen**
/cc @vangent if you're interested

Updated sample code and conformance tests to call new `Close` method.

Fixes #1492
Fixes #1485

### CODE REVIEW

**Comment by zombiezen** [comment]
One thing this is likely missing: I didn't update the `blobvar.URLOpener` type introduced in #1449 to use `Close`, which is actually how I came about this issue.

**Comment by zombiezen** [comment]
Done. This PR now fixes #1485 as well.

File: blob/blob.go [hunk 1]
```diff
@@ -774,10 +810,27 @@ func (b *Bucket) SignedURL(ctx context.Context, key string, opts *SignedURLOptio
...
+// Close releases any resources used for the bucket.
+func (b *Bucket) Close() error {
+	b.mu.Lock()
+	prev := b.closed
+	b.closed = false
+	b.mu.Unlock()
+	if prev {
+		return errClosed
+	}
+	return b.b.Close()
```
Comment (shantuo) [inline blob/blob.go]: Also set b.closed to true?

File: runtimevar/blobvar/blobvar.go [hunk 2]
```diff
@@ -245,6 +275,10 @@ func (w *watcher) WatchVariable(ctx context.Context, prev driver.State) (driver.
 
 // Close implements driver.Close.
 func (w *watcher) Close() error {
+	if w.opener != nil {
+		w.opener.decBucketRef(w.bucket)
+		w.opener = nil // Ensure that we don't call multiple times.
```
Comment (shantuo) [inline runtimevar/blobvar/blobvar.go]: Should it also close the bucket when refs is 0?

File: blob/blob.go (hunk 1 above)
Comment (zombiezen) [reply blob/blob.go]: Whoops. Thanks.

File: runtimevar/blobvar/blobvar.go (hunk 2 above)
Comment (zombiezen) [reply runtimevar/blobvar/blobvar.go]: Yes. Done.

**Pull request merged**